### PR TITLE
Fix: ignore leading space check for `null` elements in comma-spacing (fixes #3392)

### LINKED ISSUE #3392: comma-spacing conflicts with array-bracket-spacing
### Problem description

Given the following config

```json
{
    "env": {
        "es6": true
    },
    "ecmaFeatures": {
        "modules": true
    },
    "rules": {
        "comma-spacing": [ 2, { "before": false, "after": true } ],
        "array-bracket-spacing": [ 2, "always" ]
    }
}
```

and the following code:

```js
const [ , a, ...rest ] = foo;
```

then the `comma-spacing` rule warns about an extra space before to first comma in the destructuring statement.

```
foo.js
  1:9  error  There should be no space before ','  comma-spacing

✖ 1 problem (1 error, 0 warnings)
```

when I remove the leading whitespace

```js
const [, a, ...rest ] = foo;
```

then the `array-bracket-spacing` rule warns about a missing space after the opening bracket

```
foo.js
  1:7  error  A space is required after '['  array-bracket-spacing

✖ 1 problem (1 error, 0 warnings)
```

Same problem exist for array literals:

```js
const foo = [ , 1, 2 ];
```

```
foo.js
  1:15  error  There should be no space before ','  comma-spacing

✖ 1 problem (1 error, 0 warnings)
```

I’m using ESLint version 1.1.0

-----

### Possible solution

I suggest that the `comma-spacing` rule should ignore `null` elements of an `ArrayExpression` or `ArrayPattern`.

### PR DESCRIPTION
**Pull request by lo1tuma**

### CODE REVIEW

File: tests/lib/rules/comma-spacing.js [hunk 1]
```diff
@@ -81,26 +81,30 @@ ruleTester.run("comma-spacing", rule, {
...
-        {code: "var arr = [ , 2 , 3];", options: [{before: true, after: true}]},
+        {code: "var arr = [, 2 , 3];", options: [{before: true, after: true}]},
         {code: "var arr = [1 , , 3];", options: [{before: true, after: true}]},
         {code: "var arr = [1 , 2 , 3];", options: [{before: true, after: true}]},
         {code: "a , b", options: [{before: true, after: true}]},
         {code: "var arr = [,];", options: [{before: false, after: false}]},
+        {code: "var arr = [ ,];", options: [{before: false, after: false}]},
         {code: "var arr = [1,];", options: [{before: false, after: false}]},
         {code: "var arr = [,2];", options: [{before: false, after: false}]},
+        {code: "var arr = [ ,2];", options: [{before: false, after: false}]},
```
Comment (ilyavolodin) [inline tests/lib/rules/comma-spacing.js]: Shouldn't this be failing since before is set to false?

Comment (lo1tuma) [inline tests/lib/rules/comma-spacing.js]: `before` is ignored in this case because there is no array element before the comma.

Comment (ilyavolodin) [inline tests/lib/rules/comma-spacing.js]: Hmm.. That's a bit confusing. Should that be clearly documented somewhere?

Comment (lo1tuma) [inline tests/lib/rules/comma-spacing.js]: Actually there is nothing before the comment, so the rule should try to validate the spacing between nothing and a comma token.

Comment (nzakas) [inline tests/lib/rules/comma-spacing.js]: Yeah, I think we need to call this out in the documentation. I'm sure this won't be the last time someone gets confused by this. I think the implementation is fine, we just need some education around it.

Comment (lo1tuma) [inline tests/lib/rules/comma-spacing.js]: Alright, I’ve added a note to the documentation.

**Comment by ilyavolodin** [comment]
LGTM

**Pull request merged**